primitives/src/calendar.rs

```diff
@@ -57,22 +95,27 @@ pub struct CalendarPeriod {
 }
 
 impl CalendarPeriod {
-    /// For fixed length calendar periods (in Unix time, without leap seconds), computes the number
-    /// of seconds they contain. For variable length periods, returns a
-    /// `VariableLengthCalendarPeriod`.
-    pub fn as_fixed_or_variable(&self) -> FixedOrVariableCalendarUnit {
-        match self.unit {
-            CalendarUnit::Second => FixedOrVariableCalendarUnit::Fixed(self.multiplier),
-            CalendarUnit::Minute => FixedOrVariableCalendarUnit::Fixed(self.multiplier * 60),
-            CalendarUnit::Hour => FixedOrVariableCalendarUnit::Fixed(self.multiplier * 60 * 60),
-            CalendarUnit::Day => FixedOrVariableCalendarUnit::Fixed(self.multiplier * 60 * 60 * 24),
-            CalendarUnit::Week => {
-                FixedOrVariableCalendarUnit::Fixed(self.multiplier * 60 * 60 * 24 * 7)
-            }
-            CalendarUnit::Month => {
-                FixedOrVariableCalendarUnit::Variable(VariableCalendarUnit::Month)
-            }
-            CalendarUnit::Year => FixedOrVariableCalendarUnit::Variable(VariableCalendarUnit::Year),
+    /// Returns the complexity of this period.
+    ///
+    /// A non-recurring period has complexity `1`.
+    /// Recurring periods depend on how often they occur in a year.
+    /// For example, one hour is more complex than one month,
+    /// and one hour is more complex than two hours.
+    pub fn complexity(&self) -> u64 {
```

Comment:
Perhaps calling this `frequency` would be more self-explanatory.

Reply:
This is more than just a frequency. For example, the `1` and `2` constants are opinionated choices that doesn't have a lot to do with it, but does have to do with complexity.

Reply:
Could you make it less opinionated by using frequency `CalendarUnit::Year.seconds_in() / self.unit.seconds_in() / self.multiplier` and setting the frequency of a non-recurring `CalendarPeriod` to 0? That would allow you to drop `.max(2)`. I think the choice of 1 as the initial value makes this more complicated.